docs/sql-ref-syntax-aux-describe-table.md

```diff
@@ -60,8 +61,95 @@ to return the metadata pertaining to a partition or column respectively.
+| ByteType              | `{ "type" : "tinyint" }`                                                                                              |
+| ShortType             | `{ "type" : "smallint" }`                                                                                             |
+| IntegerType           | `{ "type" : "int" }`                                                                                                  |
+| LongType              | `{ "type" : "bigint" }`                                                                                               |
+| FloatType             | `{ "type" : "float" }`                                                                                                |
+| DoubleType            | `{ "type" : "double" }`                                                                                               |
+| DecimalType           | `{ "type": "decimal", "precision": p, "scale": s }`                                                                   |
+| StringType            | `{ "type" : "string" }`                                                                                               |
+| VarCharType           | `{ "type" : "varchar", "length": n }`                                                                                 |
+| CharType              | `{ "type" : "char", "length": n }`                                                                                    |
+| BinaryType            | `{ "type" : "binary" }`                                                                                               |
+| BooleanType           | `{ "type" : "boolean" }`                                                                                              |
+| DateType              | `{ "type" : "date" }`                                                                                                 |
+| TimestampType         | `{ "type" : "timestamp_ltz" }`                                                                                        |
+| TimestampNTZType      | `{ "type" : "timestamp_ntz" }`                                                                                        |
+| YearMonthIntervalType | `{ "type" : "interval", "start_unit": "<start_unit>", "end_unit": "<end_unit>" }`                                     |
+| DayTimeIntervalType   | `{ "type" : "interval", "start_unit": "<start_unit>", "end_unit": "<end_unit>" }`                                     |
+| ArrayType             | `{ "type" : "array", "element_type": <type_json>, "contains_null": <boolean> }`                                       |
+| MapType               | `{ "type" : "map", "key_type": <type_json>, "value_type": <type_json>, "contains_null": <boolean> }`                  |
+| StructType            | `{ "name" : "<name>", "type" : "struct", "nullable": <boolean>, "comment": "<comment>", "default": "<default_val>" }` |
```

Comment:
o we have array, and map with contains_null, and struct with nullable.... Why?

Reply:
the field names come from the [Spark Data Types docs page](https://spark.apache.org/docs/4.0.0-preview2/sql-ref-datatypes.html), should we use the same ones?

Reply:
Since we use the SQL type names we should stick with that. I'm ok with not_null, or nullable. Contains Null is really a bad name. Given struct has nullable (and it avpid double negation), let's go with that.